src/target/riscv/riscv_reg.c

```diff
@@ -888,6 +888,34 @@ static int riscv_set_or_write_register(struct target *target,
 	return ERROR_OK;
 }
 
+bool riscv_reg_any_dirty(const struct target *target, int log_level)
```

Comment:
Optionally, if you like, the function could be renamed to contain the word `cache`, to make it very clear to the reader what is going on.

```suggestion
bool riscv_reg_cache_any_dirty(const struct target *target, int log_level)
```

Reply:
Addressed.